### PR TITLE
Add ironicclient

### PR DESCRIPTION
**Pull request by brainstorm**
Breaking up dependencies for PR #3421 (OpenStack+Ansible)

### CODE REVIEW

**Comment by brainstorm** [comment]
Dependent on #3436 being present/uploaded into conda-forge.

File: recipes/python-ironicclient/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - python
+    - pbr
+
+  run:
+    - python
+    - pbr
+    - keystoneauth1
+    - oslo.utils
+    - oslo.i18n
+    - oslo_serialization
```
Comment (jakirkham) [inline recipes/python-ironicclient/meta.yaml]: This should have a `.` right?

File: recipes/python-ironicclient/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  skip: True  # [py3k]
+
+requirements:
+  build:
+    - python
```
Comment (jakirkham) [inline recipes/python-ironicclient/meta.yaml]: Also needs `setuptools`.

File: recipes/python-ironicclient/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  commands:
+    - ironic help
+
+about:
+  home: https://docs.openstack.org/python-ironicclient/
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE
+  summary: Python Bindings to the OpenStack Ironic API
+  dev_url: https://github.com/openstack/python-ironicclient
```
Comment (jakirkham) [inline recipes/python-ironicclient/meta.yaml]: According to PyPI the source is at ( https://git.openstack.org/cgit/openstack/python-ironicclient ).

File: recipes/python-ironicclient/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    - six
+
+test:
+  imports:
+    - ironicclient
+  commands:
+    - ironic help
+
+about:
+  home: https://docs.openstack.org/python-ironicclient/
```
Comment (jakirkham) [inline recipes/python-ironicclient/meta.yaml]: According to PyPI this is ( https://docs.openstack.org/python-ironicclient/latest/ ).

File: recipes/python-ironicclient/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    - oslo.i18n
+    - oslo_serialization
+    - prettytable
+    - six
+
+test:
+  imports:
+    - ironicclient
+  commands:
+    - ironic help
```
Comment (jakirkham) [inline recipes/python-ironicclient/meta.yaml]: Does `help` need `-` or `--` immediately before?

File: recipes/python-ironicclient/meta.yaml (hunk 1 above)
Comment (brainstorm) [inline recipes/python-ironicclient/meta.yaml]: Nope: https://docs.openstack.org/oslo.serialization/latest/reference/index.html

File: recipes/python-ironicclient/meta.yaml (hunk 5 above)
Comment (brainstorm) [inline recipes/python-ironicclient/meta.yaml]: No: https://docs.openstack.org/python-ironicclient/latest/cli/ironic_client.html#examples

**Comment by brainstorm** [comment]
I don't get the root cause of this error on TravisCI:

https://travis-ci.org/conda-forge/staged-recipes/builds/262147375#L675

The dependencies have already been uploaded to conda-forge proper:

```
$ conda search -c conda-forge oslo.serialization
Fetching package metadata .............
oslo.serialization           2.20.0                   py27_0  conda-forge
                             2.20.0                   py35_0  conda-forge
                             2.20.0                   py36_0  conda-forge
```

File: recipes/python-ironicclient/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/python-ironicclient/meta.yaml]: Why not? The name of the package on `conda-forge` is `oslo.serialization`.

File: recipes/python-ironicclient/meta.yaml (hunk 5 above)
Comment (jakirkham) [inline recipes/python-ironicclient/meta.yaml]: Cool, thanks.

**Comment by jakirkham** [comment]
Please see my comment above in the diff about `_` vs `.`.

**Comment by brainstorm** [comment]
Freenode's #openstack-ironic channel extract:

[12:23] <brainstorm_> That is, "import_versioned_module() takes at most 2 arguments (3 given)"
[12:26] \<dtantsur\> brainstorm_: hi! quick guess: old version of oslo.utils?
[12:29] <brainstorm_> dtantsur: huh, I see 3.23 vs 3.28, and it indeed seems painful to repackage due to the multiple intricated version pinnings: https://github.com/conda-forge/oslo.utils-feedstock/blob/master/recipe/meta.yaml

**Comment by brainstorm** [comment]
@jakirkham (or any of the conda-forge admins)... tests passing! Can you merge this one so that I can move on to the other two PRs (and finally ansible?).

Thanks!

**Comment by brainstorm** [comment]
@conda-forge/staged-recipes, can anyone merge this one?

**Comment by scopatz** [comment]
LGTM Thanks!

**Pull request merged**